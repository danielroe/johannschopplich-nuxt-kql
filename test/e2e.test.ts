import { fileURLToPath } from 'node:url'
import destr from 'destr'
import { describe, expect, it } from 'vitest'
import { $fetch, setup } from '@nuxt/test-utils'

describe('nuxt-kql', async () => {
  await setup({
    server: true,
    rootDir: fileURLToPath(new URL('../playground', import.meta.url)),
  })

  it('fetches queries with $kql', async () => {
    const html = await $fetch('/test/$kql')
    expect(getTestResult(html)).toMatchSnapshot()
  })

  it('fetches queries with $kql (client requests)', async () => {
    const html = await $fetch('/test/$kql_client')
    expect(getTestResult(html)).toMatchSnapshot()
  })

  it('fetches queries with useKql (client requests)', async () => {
    const html = await $fetch('/test/useKql_client')
    expect(getTestResult(html)).toMatchSnapshot()
  })

  it('fetches Kirby data useKirbyData', async () => {
    const html = await $fetch('/test/useKirbyData')
    expect(getTestResult(html)).toMatchSnapshot()
  })

  it('fetches Kirby data useKirbyData (client requests)', async () => {
    const html = await $fetch('/test/useKirbyData_client')
    expect(getTestResult(html)).toMatchSnapshot()
  })

  it('can prefetch KQL queries', async () => {
    const html = await $fetch('/test/prefetch')
    expect(getTestResult(html)).toMatchSnapshot()
  })
})

function getTestResult(html: string) {
  const content = html.match(/<script\s+type="text\/test-result">(.*?)<\/script>/s)?.[1]
  return destr(content)
}
