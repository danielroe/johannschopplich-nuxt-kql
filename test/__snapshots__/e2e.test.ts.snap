// Vitest Snapshot v1

exports[`nuxt-kql > can prefetch KQL queries 1`] = `
{
  "children": [
    {
      "id": "photography",
      "isListed": true,
      "title": "Photography",
    },
    {
      "id": "notes",
      "isListed": true,
      "title": "Notes",
    },
    {
      "id": "about",
      "isListed": true,
      "title": "About us",
    },
    {
      "id": "error",
      "isListed": false,
      "title": "Error",
    },
    {
      "id": "home",
      "isListed": false,
      "title": "Home",
    },
    {
      "id": "sandbox",
      "isListed": false,
      "title": "Sandbox",
    },
  ],
  "title": "Mægazine",
}
`;

exports[`nuxt-kql > fetches Kirby data useKirbyData (client requests) 1`] = `
{
  "children": [
    {
      "id": "photography",
      "isListed": true,
      "title": "Photography",
    },
    {
      "id": "notes",
      "isListed": true,
      "title": "Notes",
    },
    {
      "id": "about",
      "isListed": true,
      "title": "About us",
    },
    {
      "id": "error",
      "isListed": false,
      "title": "Error",
    },
    {
      "id": "home",
      "isListed": false,
      "title": "Home",
    },
    {
      "id": "sandbox",
      "isListed": false,
      "title": "Sandbox",
    },
  ],
  "title": "Mægazine",
}
`;

exports[`nuxt-kql > fetches Kirby data useKirbyData 1`] = `
{
  "children": [
    {
      "id": "photography",
      "isListed": true,
      "title": "Photography",
    },
    {
      "id": "notes",
      "isListed": true,
      "title": "Notes",
    },
    {
      "id": "about",
      "isListed": true,
      "title": "About us",
    },
    {
      "id": "error",
      "isListed": false,
      "title": "Error",
    },
    {
      "id": "home",
      "isListed": false,
      "title": "Home",
    },
    {
      "id": "sandbox",
      "isListed": false,
      "title": "Sandbox",
    },
  ],
  "title": "Mægazine",
}
`;

exports[`nuxt-kql > fetches queries with $kql (client requests) 1`] = `
{
  "code": 200,
  "result": {
    "children": [
      {
        "id": "photography",
        "isListed": true,
        "title": "Photography",
      },
      {
        "id": "notes",
        "isListed": true,
        "title": "Notes",
      },
      {
        "id": "about",
        "isListed": true,
        "title": "About us",
      },
      {
        "id": "error",
        "isListed": false,
        "title": "Error",
      },
      {
        "id": "home",
        "isListed": false,
        "title": "Home",
      },
      {
        "id": "sandbox",
        "isListed": false,
        "title": "Sandbox",
      },
    ],
    "title": "Mægazine",
  },
  "status": "OK",
}
`;

exports[`nuxt-kql > fetches queries with $kql 1`] = `
{
  "code": 200,
  "result": {
    "children": [
      {
        "id": "photography",
        "isListed": true,
        "title": "Photography",
      },
      {
        "id": "notes",
        "isListed": true,
        "title": "Notes",
      },
      {
        "id": "about",
        "isListed": true,
        "title": "About us",
      },
      {
        "id": "error",
        "isListed": false,
        "title": "Error",
      },
      {
        "id": "home",
        "isListed": false,
        "title": "Home",
      },
      {
        "id": "sandbox",
        "isListed": false,
        "title": "Sandbox",
      },
    ],
    "title": "Mægazine",
  },
  "status": "OK",
}
`;

exports[`nuxt-kql > fetches queries with useKql (client requests) 1`] = `
{
  "code": 200,
  "result": {
    "children": [
      {
        "id": "photography",
        "isListed": true,
        "title": "Photography",
      },
      {
        "id": "notes",
        "isListed": true,
        "title": "Notes",
      },
      {
        "id": "about",
        "isListed": true,
        "title": "About us",
      },
      {
        "id": "error",
        "isListed": false,
        "title": "Error",
      },
      {
        "id": "home",
        "isListed": false,
        "title": "Home",
      },
      {
        "id": "sandbox",
        "isListed": false,
        "title": "Sandbox",
      },
    ],
    "title": "Mægazine",
  },
  "status": "OK",
}
`;
